_episodes/01-why-bother-performance.md

```diff
@@ -1,16 +1,133 @@
 ---
 title: "Why bother with performance?"
-teaching: 0
-exercises: 0
+teaching: 10
+exercises: 5
 questions:
-- "Key question (FIXME)"
+- "Why is software performance important?"
+- "How can performance be measured?"
+- "What is meant by flops, walltime and cpuh?"
+- "What affects performance?"
 objectives:
-- "First learning objective. (FIXME)"
+- "Understand the necessity of code optimisation"
+- "Identify the different methods of enhancing performance"
+- "Calculate walltime, cpuh"
 keypoints:
-- "First key point. Brief Answer to questions. (FIXME)"
+- "Understanding performance is the best way of utilising your HPC resources"
+- "There are many ways of enhancing performance"
+- "Just because a way is correct doesn't always mean it is the best option"
 ---
 
-FIXME
+## What is software performance?
 
-{% include links.md %}
 
+
+## Why is software performance important?
+
+(FIXME)
+
+> ## Enchancing performance: rationale
+>
+> Imagine you had a 40 x 40 x 40 box like the one below, divided up into smaller boxes, each measuring 1 x 1 x 1. If you wanted and you wanted to simulate what was happening inside each smaller box for 10 hours each How long will the example below take to run on one core?
+>
+> include figure.html url="" max-width="40%" file="/fig/01/40x40x40_cube.png" alt="Figure" caption="40 x 40 x 40 cube"
+> 
+> {: language-bash}
+> 
+>> ## Solution
+>> 
+>> 64000 hours or just over 73 years.
+>>
+>> This is way longer than anyone could bear! But remember, that is utilising just one core. If you had a machine that could simulate each of those smaller boxes simultaneously and a code that enables each box to effectively interact with each other, the whole job would only take roughly 10-12 hours. So it provides the motivation we need to speed up the code we have and utilise the machine to its best capacity
+>>
+>{: .solution}
+{: .challenge}
+
```

Comment:
You may consider to add one short section here:
## What is performance? 
Can we explain with some very general example (not necessarily need to be connected with computing. It could be be something from our daily life so that it becomes very easy to appreciate even for a novice)

Reply:
Made a 'chef' example, with a head chef and multiple workers.